internal/querynodev2/segments/validate.go

```diff
@@ -42,21 +41,15 @@ func validate(ctx context.Context, manager *Manager, collectionID int64, partiti
 	if len(partitionIDs) == 0 {
 		searchPartIDs = collection.GetPartitions()
 	} else {
-		collection.ExistPartition()
-		if collection.ExistPartition(partitionIDs...) {
-			searchPartIDs = partitionIDs
+		if !collection.ExistPartition(partitionIDs...) {
+			return nil, nil, merr.WrapErrPartitionNotLoaded(partitionIDs)
 		}
```

Comment:
It looks like the validation cannot be performed here. Querycoord dipsatches release partition in random order, if a worker recieved release partition request before the delegator do, it might report an unwanted search error